PxGet/Functions/Invoke-PxGet.ps1

```diff
@@ -0,0 +1,152 @@
+        }
+        else 
+        {
+            Write-Error 'The pxget.json file does not exist!' -ErrorAction SilentlyContinue
+            return
+        }
+        
+        if( $pxModules )
+        {
+            $moduleNames = $pxModules.PSModules | Select-Object -ExpandProperty 'Name'
+            if( $moduleNames )
+            {
+                $modules = Find-Module -Name $moduleNames -ErrorAction Ignore
+
+                # We only care if the module is in PSModules right now. Later we'll allow dev dependencies, which can be
+                # installed globally.
+                $env:PSModulePath = $psModulesPath
+                foreach( $pxModule in $pxModules.PSModules )
+                {
+                    $allowPrerelease = $pxModule.Version.Contains('-')
+                    
+                    if( $allowPrerelease)
+                    {
+                        # Find module again but with AllVersions and AllowPrerelease
+                        $modulesWithPrelease = Find-Module -Name $pxModule.Name -AllowPrerelease -AllVersions
+                        $moduleToInstall = FindModuleFromList -Modules $modulesWithPrelease -ModuleToFind $pxModule -AllowPrerelease
+                    }
+                    else
+                    {
+                        if( -not $modules )
+                        {
+                            Write-Error 'No modules were found using the module names from the pxget file!' -ErrorAction SilentlyContinue
+                            continue
+                        }
+                        $moduleToInstall = FindModuleFromList -Modules $modules -ModuleToFind $pxModule
+                    }
+        
+                    if( -not $moduleToInstall )
+                    {
+                        Write-Error "Module $($pxModule.Name) was not found!"
+                        continue
+                    }
+        
+                    $installedModule =
+                        Get-Module -Name $pxModule.Name -List |
+                        Where-Object 'Version' -eq $moduleToInstall.Version
+                    # The latest version that matches the version in the pxget.json file is already installed
+                    if( $installedModule )
+                    {
+                        continue
+                    }
+        
+                    # Not installed. Install it. We pipe it so the repository of the module is also used.
+                    $moduleToInstall | Save-Module -Path $psmodulesPath
+                }
+            }
+            else 
+            {
+                Write-Warning 'There are no modules listed in the pxget.json file!'
+            }
+        }
+        else 
+        {
+            Write-Warning 'The pxget.json file is empty!'
+        }
+    }
+    finally
+    {
+        $env:PSModulePath = $origModulePath
+    }
+}
+
+Set-Alias -Name 'pxget' -Value 'Invoke-PxGet'
+
+function FindModuleFromList

```

Comment:
Rename to `Select-Module`.

Change this to a function that accepts pipeline input [1] then update usages to pipe in the modules to search: `$moduleToInstall = $modules | Select-Module -Name $pxmodule.Name -Version $pxmodule.Version`.

[1]

```powershell
function Select-Module
{
    [CmdletBinding()]
    param(
        [Parameter(Mandatory)]
        [Microsoft.PowerShell.Commands.PSRepositoryItemInfo] $Module,

        [Parameter(Mandatory)]
        [String] $Name,

        [Parameter(Mandatory)]
        [String] $Version,  

        [switch] $AllowPrerelease
    )

    process
    {
        if( $module.Name -ne $Name -or $module.Version -notlike $Version )
        {
            return
        }

        if( $AllowPrerelease )
        {
            return $Module
        }

        [Version]$moduleVersion = $null
        if( [Version]::TryParse($Module.Version, [ref]$moduleVersion) )
        {
            return $Module
        }
    }
}
```